src/Bridges/Constraint/map.jl

```diff
@@ -0,0 +1,188 @@
+struct Map <: AbstractDict{MOI.VariableIndex, AbstractBridge}
+    # Constraint Index of bridged constraint -> Bridge.
+    # It is set to `nothing` when the constraint is deleted.
+    bridges::Vector{Union{Nothing, AbstractBridge}}
+    # Constraint Index of bridged constraint -> Constraint type.
+    constraint_types::Vector{Tuple{DataType, DataType}}
```

Comment:
Ditto here. I think we can do better than `DataType`, if only for readability sake.

```Julia
const TupleConstraintType = Tuple{<:MOI.AbstractFunction, <:MOI.AbstractSet}

constraint_types::Vector{TupleConstraintType}
```

Reply:
This doesn't work because the elements are not functions, they are function *types*.
with your code, I get
```julia
julia> constraint_types[1] = (MOI.SingleVariable, MOI.EqualTo{Float64})
ERROR: MethodError: Cannot `convert` an object of type Type{MathOptInterface.SingleVariable} to an object of type MathOptInterface.AbstractFunction
Closest candidates are:
  convert(::Type{T}, ::T) where T at essentials.jl:154
Stacktrace:
 [1] convert(::Type{Tuple{#s2,#s1} where #s1<:MathOptInterface.AbstractSet where #s2<:MathOptInterface.AbstractFunction}, ::Tuple{DataType,DataType}) at ./essentials.jl:275
 [2] setindex!(::Array{Tuple{#s2,#s1} where #s1<:MathOptInterface.AbstractSet where #s2<:MathOptInterface.AbstractFunction,1}, ::Tuple{DataType,DataType}, ::Int64) at ./array.jl:767
 [3] top-level scope at none:0
```